src/core/services/layout/layouts.spec.js

```diff
@@ -0,0 +1,87 @@
+                        { attribute: 'offset', suffixes: suffixes, values: offsetValues },
+                        { attribute: 'layout-padding', testStandAlone: true },
+                        { attribute: 'layout-margin', testStandAlone: true },
+                        { attribute: 'layout-wrap', testStandAlone: true },
+                        { attribute: 'layout-fill', testStandAlone: true },
+                        { attribute: 'hide', suffixes: suffixes, testStandAlone: true },
+                        { attribute: 'show', suffixes: suffixes, testStandAlone: true }];
+
+        function testMapping(attribute, expectedClass) {
+            it('should fail if the class ' + expectedClass + ' was not added for attribute ' + attribute, inject(function ($compile, $rootScope) {
+                var element = $compile('<div ' + attribute + '>Layout</div>')($rootScope);
+                expect(element.hasClass(expectedClass)).toBe(true);
+            }));
+        }
+
+        function testWithSuffix(attribute, suffixes, values, testStandAlone, addDirectiveAsClass) {
+            for (var j = 0; j < suffixes.length; j++) {
+                var suffix = suffixes[j];
+                // Add the suffix to the attribute.
+                var attributeWithValue = attribute + '-' + suffix;
+                // Add the suffix to the expected class.
+                var expectedClass = attribute + '-' + suffix;
+                // Run the test.
+                if (testStandAlone)
+                    testMapping(attributeWithValue, expectedClass);
+                // Add suffixes with values.
+                if (values)
+                    testWithSuffixAndValue(attribute, values, suffix, addDirectiveAsClass);
+            };
+        }
+
+        function testWithSuffixAndValue(attribute, values, suffix, addDirectiveAsClass) {
+            for (var j = 0; j < values.length; j++) {
+                var attributeValue = values[j].toString();
+                // If there was a suffix passed in, add it first.
+                var attributeWithValue = suffix ? attribute + '-' + suffix : attribute;
+                // Add the value.
+                attributeWithValue += '="' + attributeValue + '"';
+                // The expected class always starts with the attribute name, add the suffix if any.
+                var expectedClass = suffix ? attribute + '-' + suffix : attribute;
+                // Add the class if necessary.
+                if (addDirectiveAsClass)
+                    expectedClass += ' ' + expectedClass;
+                // Add the attribute and suffix to the expected class and replace the spaces with a dash.                
+                expectedClass += '-' + attributeValue.replace(/\s+/g, "-");
+                // Run the test.
+                testMapping(attributeWithValue, expectedClass);
+            };
+        }
+
+        for (var i = 0; i < mappings.length; i++) {
```

Comment:
Let's move this **for** loop block to line 32... so the function definitions will follow subsequently.